client/src/containers/charts/bar.tsx

```diff
@@ -0,0 +1,73 @@
+import React from 'react';
+
+import { BarChart, Bar, Rectangle, XAxis, YAxis, ResponsiveContainer } from 'recharts';
+
+const CustomizedXAxisTick = ({
+  x,
+  y,
+  payload,
+}: {
+  x: number;
+  y: number;
+  payload: { value: string };
+}) => {
+  return (
+    <g transform={`translate(${x},${y})`}>
+      <text x={0} y={0} dy={16} textAnchor="middle" fill="#7C828A" fontSize={12}>
+        {payload.value}
+      </text>
+    </g>
+  );
+};
+
+const CustomizedYAxisTick = ({
+  x,
+  y,
+  payload,
+}: {
+  x: number;
+  y: number;
+  payload: { value: string };
+}) => {
+  return (
+    <g transform={`translate(${x},${y})`}>
+      <text x={0} y={0} dy={0} dx={-2} textAnchor="middle" fill="#7C828A" fontSize={12}>
+        {payload.value}
+      </text>
+    </g>
+  );
+};
+
+export default function BarChartComponent({
+  data,
+  margin,
+}: {
+  data: { [key: string]: unknown }[];
+  margin: { top: number; right: number; bottom: number; left: number };
+}) {
+  return (
+    <ResponsiveContainer width="100%" height="100%">
+      <BarChart width={500} height={300} data={data} margin={margin}>
+        <XAxis
+          dataKey="year"
+          axisLine={false}
+          tickLine={false}
+          ticks={['2000', '2007', '2014']}
+          tick={CustomizedXAxisTick}
+        />
+        <YAxis
+          axisLine={false}
+          tickLine={false}
+          ticks={['0', '50', '100']}
+          tick={CustomizedYAxisTick}
+        />
+        <Bar
+          dataKey="uv"
+          fill="#70CCB0"
+          radius={[2, 2, 0, 0]}
+          activeBar={<Rectangle stroke="yellow" />}
+        />
+      </BarChart>
+    </ResponsiveContainer>
+  );
+}
```

Comment:
I'd made this component more flexible, instead of adding the make elements consume a configuration and adding specific props according to that configuration